packages/inngest/src/components/Inngest.ts

```diff
@@ -561,26 +565,36 @@ export class Inngest<TClientOpts extends ClientOptions = ClientOptions> {
       }
     }
 
-    // We don't need to do fallback auth here because this uses event keys and
-    // not signing keys
-    const response = await this.fetch(url, {
-      method: "POST",
-      body: stringify(payloads),
-      headers: { ...this.headers, ...headers },
-    });
-
-    let body: SendEventResponse | undefined;
-
-    try {
-      const rawBody: unknown = await response.json();
-      body = await sendEventResponseSchema.parseAsync(rawBody);
-    } catch (err) {
-      throw await this.getResponseError(response);
-    }
-
-    if (body.status / 100 !== 2 || body.error) {
-      throw await this.getResponseError(response, body.error);
-    }
+    const body = await retryWithBackoff(
+      async () => {
+        let body: SendEventResponse | undefined;
+
+        // We don't need to do fallback auth here because this uses event keys and
+        // not signing keys
+        const response = await this.fetch(url, {
+          method: "POST",
+          body: stringify(payloads),
+          headers: { ...this.headers, ...headers },
+        });
+
+        try {
+          const rawBody: unknown = await response.json();
+          body = await sendEventResponseSchema.parseAsync(rawBody);
+        } catch (err) {
+          throw await this.getResponseError(response);
+        }
+
+        if (body.status !== 200 || body.error) {
+          throw await this.getResponseError(response, body.error);
+        }
+
+        return body;
+      },
+      {
+        maxAttempts: 5,
```

Comment:
Future - would be nice to configure this optionally like `inngest.send(payload, { maxAttempts: 3 })`

Reply:
💯 as well as customizing backoff strategy!